source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -286,15 +301,10 @@ namespace PlayFab
     {
         try
         {
-            // batch was successfully sent out, find it in the batch tracking map using customData pointer as a key
-            const auto foundBatchIterator = this->batchesInFlight.find(customData);
-            if (foundBatchIterator == this->batchesInFlight.end())
-            {
-                LOG_PIPELINE("Untracked batch was returned to EventsAPI.WriteEvents callback"); // normally this never happens
-            }
-            else
+            std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>> batchWritten;
+            if(TryGetBatchOutOfFlight(customData, &batchWritten))
             {
-                auto requestBatchPtr = std::shared_ptr<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(new std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>(std::move(foundBatchIterator->second)));
+                auto requestBatchPtr = std::shared_ptr<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(new std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>(batchWritten));
```

Comment:
these requestBatchPtr lines are a doozy to read.
you can make them shorter (by like 60 chars) if you use std::make_shared

```
auto requestBatchPtr = std::make_shared<std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(std::move(batchWritten));
```